services/QuillLMS/app/serializers/application_serializer.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ApplicationSerializer < ActiveModel::Serializer
```

Comment:
How can we avoid developers accidentally subclassing from ActiveModel::Serializer instead of this new class?

One way would be to monkeypatch `ActiveModel::Serializer` and include a `self.inherited` hook that raises an exception or prints a warning when `ActiveModel::Serializer` is subclassed. I don't think this is ideal, however, because it breaks the abstraction layer between the gem and LMS code.

Reply:
It's an interesting question but one that we can probably leave up to code review since this would be more about enforcing a coding convention than the code not working.  Analogous issues exists not inheriting from `ApplicationRecord` for models, `ApplicationController` and controllers, etc.  In all of these cases, there's probably some missing functionality that will nudge the developer to inherit from the correct superclass.